tests/Unit/DB/DBHelperTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\DB;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\DB\DBHelper;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\ContainerAwareUnitTest;
+use wpdb;
+
+/**
+ * Class DBHelperTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\DB
+ *
+ * @property wpdb     $wpdb
+ * @property DBHelper $db_helper
+ * @property string   $table_name
+ */
+class DBHelperTest extends ContainerAwareUnitTest {
+
+	public function test_index_exists() {
+		$this->wpdb->query( "CREATE TABLE `{$this->table_name}` (`id` bigint(20) NOT NULL);" );
```

Comment:
I've removed the DBHelper but for future reference, I couldn't find a way to easily do a `table_exists` check because WordPress unit tests hook into the DB functions and modify the create queries. This means that unit tests always create `TEMPORARY` tables. The `SHOW TABLES LIKE` query will not return those temporary tables so the `table_exists` will always return false. 

I think there might be a way around this by figuring out how WP handles it and overriding the hooks but since this issue was a bit high priority I didn't go down that route.